### ISSUE TITLE
Capturing using gstreamer and pipewiresrc no longer works

### ISSUE THREAD

**Issue by safijari** [ValveSoftware/gamescope#900]
I'm one of the developers on Decky Recorder which is a very popular plugin used to record gameplay on the Deck. I noticed that I am unable to capture the screen using gstreamer in SteamOS main. Here's versions of the relevant packages. They all seem to be the latest versions under the jupiter-main repo. It appears to be an issue with negotiation. This same setup works in stable and beta, just not in main.

```
gamescope-3.12.0.beta9-5-x86_64.pkg.tar.zst
pipewire 1:0.3.62.2.dv-1
gst-plugin-pipewire 1:0.3.62.2.dv-1
```

Additional deps needed: `gstreamer-vaapi gst-plugin-pipewire gst-plugins-bad-libs gst-plugins-good`
Command: `gst-launch-1.0 -e -vvv pipewiresrc do-timestamp=true ! vaapipostproc ! queue ! vaapih264enc ! h264parse ! matroskamux name=sink ! filesink location=/tmp/test.mkv`

Result:
```
Setting pipeline to PAUSED ...
Failed to set pipeline to PAUSED.
Setting pipeline to NULL ...
Freeing pipeline ...
```

(I'll submit debug logs in a comment)

**Comment by safijari** [comment]
Debug logs
[gstreamer-deck-logs.txt](https://github.com/ValveSoftware/gamescope/files/11762488/gstreamer-deck-logs.txt)

**Comment by safijari** [comment]
Looks like there's a movement on this possibly related issue

https://github.com/ValveSoftware/gamescope/issues/775#issuecomment-1620330060

**Comment by safijari** [comment]
I'm going to commit a cardinal sin here and nudge @Joshua-Ashton on this who appears to be the most active contributor to the project right now. As it stands this is still broken on the latest preview build on the Deck. Is there any chance someone who knows what they're doing can please look into it?

I'm also happy to contribute here but I don't know how. I have an environment set up where I can build gamescope and gstreamer and I can reproduce this issue there with a simple test, but I'm completely clueless as to where I should go digging to try to solve this bug.

**Comment by mozzenior** [comment]
FWIW, in my case I failed with the below command. (My purpose was just to make gst-launch-1.0 run.)
```
$ gst-launch-1.0 pipewiresrc ! fakesink
Setting pipeline to PAUSED ...
Failed to set pipeline to PAUSED.
Setting pipeline to NULL ...
Freeing pipeline ...
```

With pipewire logs turned on I found this warning suspicious.
```
[W][12563.666058] pw.stream    | [        stream.c: 1770 find_format()] 0x562332e5afd0: no format given
```

After replacing `fakesink` with `fakevideosink`, my issue is solved.

**Comment by darkuni** [comment]
It is very common for consoles (and PCs!) to have an innate means of recording video during play.  Obviously, we would prefer official support, but in the meantime?  Let's get this plugin working in 3.5 sooner than later.

Much appreciated.

**Comment by nshank63** [comment]
Could someone at Valve please look into this

**Comment by Joshua-Ashton** [comment]
I can take a look -- I will say though that Steam In-home Streaming still appears to be functional so I find it kinda strange.

I've been meaning to upstream my OBS patches for Gamescope support for a while, so I guess I can see if there's issues there too.

**Comment by safijari** [comment]
Thank you so much @Joshua-Ashton. As mentioned I'm happy to help with debugging and testing this I just don't know where to start. Best as I can tell something in the negotiation step fails.

**Comment by Joshua-Ashton** [comment]
This should be fixed in the 3.5 release today. It wasn't a Gamescope bug. It was happening because we added a virtual audio source thing and it was technically paused sometimes and gstreamer was doing something and urgh. We just ended up removing the virtual source for now.

**Comment by parkerlreed** [comment]
3.5 is Beta right? I'll be sure to keep an eye out for 3.6 on Main! Thank you!
